### PR TITLE
Feature/skip cinematics

### PR DESCRIPTION
**Pull request by marcorod94**
Skip cinematics using punch button J\B

Known hiccup the skip UI may not appear on the first run sometimes

### CODE REVIEW

File: Scripts/src/Script/CinematicCamera.cpp [hunk 1]
```diff
@@ -303,11 +310,23 @@ void CinematicCamera::Update()
...
 		//Retrieve next element
 		current_action = cinematic_queue.front();
+		ui_manager->ShowSkipButton();
+		player_1->player_controller->player_attack->is_attack_blocked = true;
 	}
 
 	//If action we call update
 	if(current_action)
 	{
+		if (input_manager->GetGameInputDown("Punch", static_cast<PlayerID>(player_1->player_controller->player - 1)))
```
Comment (Dartemiss) [inline Scripts/src/Script/CinematicCamera.cpp:320]: Only player1 can skil cinematics?

File: Scripts/src/Script/UIManager.cpp [hunk 2]
```diff
@@ -399,9 +399,41 @@ void UIManager::InitPublicGameObjects()
...
+	selected_skip_button->SetEnabled(true);
+}
+
+void UIManager::HideSkipButton()
+{
+	if (selected_skip_button != nullptr)
+	{
+		selected_skip_button->SetEnabled(false);
+	}
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/UIManager.cpp:427]: const

File: Scripts/src/Script/UIManager.cpp [hunk 3]
```diff
@@ -399,9 +399,41 @@ void UIManager::InitPublicGameObjects()
...
+{
+	if (input_manager->total_game_controllers == 0)
+	{
+		return skip_button_keyboard;
+	}
+	else
+	{
+		return skip_button_controller;
+	}
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/UIManager.cpp:439]: const

**Review by Dartemiss — CHANGES_REQUESTED** [review]
Works fine but I haven't seen any UI at all, even restarting. Also some const correctness

File: Scripts/src/Script/CinematicCamera.cpp (hunk 1 above)
Comment (marcorod94) [reply Scripts/src/Script/CinematicCamera.cpp:320]: Yes only the player1 could do that

**Review by yalania — COMMENTED** [review]
I tried it but it doesn't work for me, the skip ui stay on the middle of the screen (I tried J in keyboard and B in controller)

**Review by yalania — APPROVED** [review]
now it works fine

**Review by yalania — CHANGES_REQUESTED** [review]
I tried second cinematic, and when you skip it continues playing at the moment it was skip.

**Comment by marcorod94** [comment]
Everything done please check it

**Comment by Dartemiss** [comment]
This uses old cinematic system so I close this.

**Pull request closed without merging**